frontend/src/components/MapPopupContent/MapPopupContent.js

```diff
@@ -1,62 +1,69 @@
-import React from 'react'
-import { Box, Button, Typography } from '@mui/material'
+import React, { useMemo } from 'react'
+import { Box, Button, ThemeProvider, Typography } from '@mui/material'
 import PropTypes from 'prop-types'
 import SimpleRange from '../SimpleRange'
 import CompareIconPlus from '../Icons/CompareIconPlus'
+import initTheme from '../../utils/theme'
 
 function MapPopupContent({ id, title, metricName, data, colorScheme, domain, expanded }) {
+  // TODO: use darkTheme from Redux
+  const darkTheme = false
+  const theme = useMemo(() => initTheme(darkTheme), [darkTheme])
+
   return (
-    <Box
-      p={1.5}
-      bgcolor={expanded ? '#fff' : 'rgba(255, 255, 255, 0.9)'}
-      color={'#333'}
-      borderRadius={'10px'}
-      width={200}
-      boxShadow={'0px 2px 4px rgba(0, 0, 0, 0.25)'}>
-      <Typography fontSize={18} fontWeight={500} color={'#000'} mb={'8px'}>
-        {title}
-      </Typography>
-      <Typography fontSize={14} mb={0.5}>
-        {metricName}
-      </Typography>
+    <ThemeProvider theme={theme}>
```

Comment:
Adding a new ThemeProvider for MapPopupContent is unnecessary. The underlying problem is the use of `ReactDOM.render` in Map.js, which has another undesirable side effect beyond disconnecting the popup from the main ThemeProvider. It leaks components. A new MapPopupContent is created every time `hoveredFeatureId` changes or a popup is expanded. Using a Portal would solve both of those problems and obviate the need for a second ThemeProvider here.

It's easier to implement than explain, so I'll push a commit which does it shortly.